ribbon/sample.ipynb

```diff
@@ -0,0 +1,336 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "types = {\n",
+    "    'Bid': [\n",
+    "      {'name': 'swapId', 'type': 'uint256'},\n",
+    "      {'name': 'nonce', 'type': 'uint256'},\n",
+    "      {'name': 'signerWallet', 'type': 'address'},\n",
+    "      {'name': 'sellAmount', 'type': 'uint256'},\n",
+    "      {'name': 'buyAmount', 'type': 'uint256'},\n",
+    "      {'name': 'referrer', 'type': 'address'},\n",
+    "    ]\n",
+    "  }\n",
+    "\n",
+    "good_input = {\n",
+    "  'swapId': 1,\n",
+    "  'nonce': 1,\n",
+    "  'signerWallet': '0x28b2732460D09a8CDdd66cE956bE41D35B12F3b1',\n",
+    "  'sellAmount': '6000000',\n",
+    "  'buyAmount': '1000000000000000000',\n",
+    "  'referrer': '0x0000000000000000000000000000000000000000'\n",
+    "}\n",
+    "\n",
+    "bad_input = {\n",
+    "  'swapId': 1,\n",
+    "  'nonce': 1,\n",
+    "  'signerWallet': '0xF49e9fc4487E1B23F92bC8effde5010e91b706eF',\n",
+    "  'sellAmount': '1000',\n",
+    "  'buyAmount': '1000000000000000',\n",
+    "  'referrer': '0x0000000000000000000000000000000000000000',\n",
+    "}\n",
+    "\n",
+    "domain = {\n",
+    "  'name': 'RIBBON SWAP',\n",
+    "  'version': '1',\n",
+    "  'chainId': 42,\n",
+    "  'verifyingContract': '0x58848824baEb9678847aF487CB02EAba782FECB5'\n",
+    "}\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'swapId': 1,\n",
+       " 'nonce': 1,\n",
+       " 'signerWallet': '0x28b2732460D09a8CDdd66cE956bE41D35B12F3b1',\n",
+       " 'sellAmount': '6000000',\n",
+       " 'buyAmount': '1000000000000000000',\n",
+       " 'referrer': '0x0000000000000000000000000000000000000000',\n",
+       " 'v': 27,\n",
+       " 'r': '0xd48860fab24673d45a03d58428f36bd7d62ac115972bd2a94e040503415a9478',\n",
+       " 's': '0x32eed933d6532dc613e3167a5e839bce2c1d577b3c4b2c73eea7411fec1c9a53'}"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "privKey = os.environ.get('PRIVATE_KEY')\n",
+    "signer = SignatureGenerator(privKey)\n",
+    "bid = signer.sign(domain, types, good_input)\n",
+    "bid"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<h1> GET OTOKEN DETAILS </h1>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rpc = os.environ.get('RPC')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from option import oTokenFactory"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'collateralAsset': '0x0000000000000000000000000000000000000000',\n",
```

Comment:
In my test locally, I've got the same output, which looks like all just zeros.
Does this mean it's not yet working?

Reply:
This is the default output in Kovan.

Reply:
Ok cool!

Reply:
Are you able to provide a real example output that we can use in testing our development?